# 2 Numbers

Solution of excercise 2

1. Remember the format() function we have used in the last video?? Now pass 145 and ‘o’ in the function and see what it returned. Try to find out which representation it is.


In [ ]:
# format function to formate the number
format(145, 'o')

'221'

In the video we did use the 'b' at the second place which stand for `binary representation` of the number.

In this eample 'o' stands for `octal representation` of the number.

'221' is octal representation of 145. Now go to Google and search octal form of 145 & check if it is correct or not.

2. There is a circular pond in a village. This pond has a radius of 84meter. Can you find the area of the pond? (Bonus: If there is exactly 1.4 liter of water in a square meter, what is the total amount of the water in the pond?)

    Hint:   
`Circle Area = πr2`


In [ ]:
# circular pond
radius = 84


As we know the formula for calculating the area of circle we can directly apply it in python

NOTE: In python ** is used for power operator


In [ ]:
pond_area = 3.14 * radius ** 2


We know that value of pi(π) is 22/7 which is equal to 3.14

In [ ]:
print(pond_area)

22155.84


**BONUS**

Because there is exactly 1.4 liter of water in 1 square meter. And we know that the total area of pond is `22155.84` square meter.

We can get the total amount of water if we multiply the given amount wih given area. 

In [ ]:
per_sqt_meter_water = 1.4

In [ ]:
total_water = pond_area * per_sqt_meter_water

In [ ]:
print(total_water)

31018.176


There is total `31018.176` liter water in the pond.

See, Programming is really fun you just calculated the total water in your village's pond. 😂

3. 
If you cross a 490-meter-long street in 7 minutes, then what is your speed in meter per seconds. Print your answer without any decimal point in it.
        Hint:
        Speed = Distance / Time


In [ ]:
street_lenght = 490
time = 7 * 60

Because Time is given in minutes ie. `7 minurtes` and we have to calculate the speed in meter pre seconds. so we need to convert the time into seconds.

We know there are 60 seconds in one minute. we can calculate the total seconds in 7 minutes by simply muliply it with 60

Let's see how many seconds are in 7 minutes

In [ ]:
time

420

there are total 420 seconds

In [ ]:
speed = street_lenght / time

Applying the formula to calculate the speed

In [ ]:
print(speed)

1.1666666666666667


The speed is 1.6666... meter per second

A lot of values after decimal, right?

let's print it without decimal

In [ ]:
print(round(speed))

1


Thank you,

Happy Coding!